course3/homework-unique_ptr/unique_ptr.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstddef>
+
+
+namespace karun
+{
+	template <class T>
+	class unique_ptr
+	{	
+		
+	public:
+		unique_ptr();
+		unique_ptr(std::nullptr_t);
```

Comment:
The 'explicit' specifier prevent implicit conversion from another type into your type. In this case the conversion is from nullptr_t to pointer.